src/vm_cpu.rs

```diff
@@ -96,6 +98,168 @@ pub struct VmCpuTrapState {
     pub htinst: u64,
 }
 
+#[derive(Default, Copy, Clone)]
+#[repr(C)]
```

Comment:
Does this (and the other structs below which were added) really need to `repr(C)`?

Reply:
Doesn't anything that's part of VmCpuState need  `repr(C)` since it crosses the ABI boundary?

Reply:
Why do the PMU bits need to be part of `VmCpuState`? Can they be a separate field in `VmCpu`?

(I had originally intended `VmCpuState` to be for the bits that we access from asm; it could probably use a better name.)

Reply:
Ah...hadn't thought about it that way...let me take a look.